jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/UpdateChecker.java

```diff
@@ -123,11 +127,24 @@ void setDisableUpdateCheck(boolean disableUpdateCheck) {
         // Generate config file if it doesn't exist
         ConfigJsonTemplate config = new ConfigJsonTemplate();
         Files.createDirectories(configDir);
-        try (OutputStream outputStream = Files.newOutputStream(configFile)) {
+        Path temp = configDir.resolve(CONFIG_FILENAME + ".tmp");
+        try (OutputStream outputStream = Files.newOutputStream(temp)) {
           JsonTemplateMapper.writeTo(config, outputStream);
+          // Attempts an atomic move first, and falls back to non-atomic if the file system does not
+          // support atomic moves.
+          try {
+            Files.move(
+                temp,
+                configFile,
+                StandardCopyOption.ATOMIC_MOVE,
+                StandardCopyOption.REPLACE_EXISTING);
+          } catch (AtomicMoveNotSupportedException ignored) {
+            Files.move(temp, configFile, StandardCopyOption.REPLACE_EXISTING);
```

Comment:
I think we don't need to do this manually. I believe `Files.move` will do this fallback.

Reply:
We do this in a few places..? CacheStorageWriter mostly? but that could be an artifact from before we knew what was happening? https://github.com/GoogleContainerTools/jib/blob/master/jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageWriter.java#L140

Reply:
Ah, sorry. Now you said it. I think this makes sense. `AtomicMoveNotSupportedException` is expected because we use the `ATOMIC_MOVE` option.

So perhaps doing `Files.move` without `ATOMIC_MOVE` is sufficient. But honestly I don't have confidence.

Reply:
Let's keep the current code? At least it seems safe.

Reply:
Which current code? This or the code before this PR?

Reply:
Sorry, the PR with two `Files.move()` that we also do in other places.